Realm/RLMSyncConfiguration.mm

```diff
@@ -270,32 +285,43 @@ - (BOOL)enableFlexibleSync {
     return make_sync_error(errorClass, @(error.message.c_str()), error.error_code.value(), custom);
 }
 
-static void setDefaults(SyncConfig& config, RLMUser *user) {
-    config.client_resync_mode = ClientResyncMode::Manual;
-    config.stop_policy = SyncSessionStopPolicy::AfterChangesUploaded;
-
+- (void)setErrorHandler:(RLMUser *)user {
     RLMSyncManager *manager = [user.app syncManager];
     __weak RLMSyncManager *weakManager = manager;
-    config.error_handler = [weakManager](std::shared_ptr<SyncSession> errored_session, SyncError error) {
+    _config->error_handler = [weakManager, self](std::shared_ptr<SyncSession> errored_session, SyncError error) {
         RLMSyncErrorReportingBlock errorHandler;
         @autoreleasepool {
             errorHandler = weakManager.errorHandler;
         }
-        if (!errorHandler) {
-            return;
-        }
         NSError *nsError = RLMTranslateSyncError(std::move(error));
         if (!nsError) {
             return;
         }
+        if (!errorHandler && nsError.code != RLMSyncErrorClientResetError) {
```

Comment:
MaybeI'm not seen it but both this conditionals seems unnecessary

Reply:
Something like
```
if (!errorHandler && !_manualClientResetHandler) {
            return;
        }

        RLMSyncSession *session = [[RLMSyncSession alloc] initWithSyncSession:errored_session];
        dispatch_async(dispatch_get_main_queue(), ^{
            // Keep the SyncSession alive until the callback completes as
            // RLMSyncSession only holds a weak reference
            static_cast<void>(errored_session);
            if (_manualClientResetHandler && nsError.code == RLMSyncErrorClientResetError) {
                self.manualClientResetHandler(nsError, session);
            } else {
                errorHandler(nsError, session);
            }
        });
```
Would crash on  errorHandler(nsError, session); for sync errors other than client reset, I think.